internal/mutable/numericarray.go

```diff
@@ -106,9 +107,10 @@ func (b *Int64Array) Reserve(n int) {
 		if b.data == nil {
 			b.init()
 		}
-		capacity := arrow.Int64Traits.BytesRequired(n)
+		length := len(b.rawData) + n
+		capacity := arrow.Int64Traits.BytesRequired(length)
```

Comment:
I see, `data.Reserve(...)` is not relative. It's builders that make it relative 👍